cmd/jaeger/internal/sanitizer/negative_duration_santizer.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025 The Jaeger Authors.
+
+package sanitizer
+
+import (
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	minDuration = time.Duration(1)
+)
+
+// NewNegativeDurationSanitizer returns a sanitizer function that performs the
+// following sanitizationson the trace data:
+//   - Checks all spans in the ResourceSpans to ensure that the start timestamp
+//     is less than the end timestamp. If not, it sets the end timestamp to
+//     start timestamp plus a minimum duration of 1 nanosecond.
+func NewNegativeDurationSanitizer() Func {
+	return sanitizeNegativeDuration
+}
+
+func sanitizeNegativeDuration(traces ptrace.Traces) ptrace.Traces {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		scopeSpans := resourceSpan.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			scopeSpan := scopeSpans.At(j)
+			spans := scopeSpan.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				sanitizeDuration(&span)
+			}
+		}
+	}
+
+	return traces
+}
+
+func sanitizeDuration(span *ptrace.Span) {
+	start := span.StartTimestamp().AsTime()
+	end := span.EndTimestamp().AsTime()
+	if start.Compare(end) == -1 {
+		return
+	}
+
+	span.SetEndTimestamp(pcommon.NewTimestampFromTime(start.Add(minDuration)))
```

Comment:
We don't destroy input data, the invalid value should be stored as a warning attribute (see jptrace/warning)